### ISSUE TITLE
Referenced string is not real string (cannot be used as dict key, doesn't equal to string)

### ISSUE THREAD

**Issue by adaamz** [koxudaxi/datamodel-code-generator#638]
**Describe the bug**

**To Reproduce**

Example schema:
```json
openapi: "3.0.0"
info:
  version: 1.0.0
  title: File endpoint
paths:
 get_file:
    parameters:
      - $ref: "#/components/schemas/FileRequest"

components:
  schemas:
    FileHash:
      type: string
      min_length: 32
      max_length: 32
      regex: "^[a-fA-F\d]{32}$"
    FileRequest:
      type: object
      required:
        - file_hash
      properties:
        file_hash:
          $ref: "#/components/schemas/FileHash"
```

Used commandline:
```sh
$ datamodel-codegen file.yml
```

Generated python code:
```py
class FileHash(BaseModel):
    __root__: str = Field(..., max_length=32, min_length=32, regex="^[0-9a-fA-F]+$")

class FileRequest(BaseModel):
    file_hash: FileHash
```

Python usage
```py
file_hash = "a"*32
f = FileRequest(file_hash=file_hash)

assert f.file_hash == file_hash) # does not equal

d = dict()
d[f.file_hash] = "asdf" # throws TypeError: unhashable type: 'FileHash'
```

**Expected behavior**
- original string equals with field
- can be used as key in dict

**Version:**
 - OS: linux
 - Python version: 3.9
 - datamodel-code-generator version: 0.11.14

**Comment by koxudaxi** [comment]
@adaamz 
I'm sorry for the slow response.
We should use `__root__` attribute for usecase.
```
assert f.file_hash.__root__ == file_hash # does not equal

d = dict()
d[f.file_hash.__root__] = "asdf" # throws TypeError: unhashable type: 'FileHash'
```

**Comment by adaamz** [comment]
yeah, i know this "workaround"... but is it possible to use the property as standard string? 🤔

**Comment by adaamz** [comment]
If I can choose combination then it would be probably best 😁 

```py
FileHash = str

class FileRequest(BaseModel):
    file_hash: FileHash = Field(...)
```

**Comment by rbuckland** [comment]
@koxudaxi  is this something you are looking at introducing ?  a feature 

effectively collapsing single primitive types, into their "base" type on the field, in a parent class

**Comment by koxudaxi** [comment]
I have released the new version as 0.14.1.
The version has this option.
```
 --collapse-root-models
                        Models generated with a root-type field will be
                        merged into the models using that root-type model
```
